### ISSUE TITLE
Example request : VICRegLoss

### ISSUE THREAD

**Issue by stereomatchingkiss** [KevinMusgrave/pytorch-metric-learning#461]
Any plan to include example of this loss function? Thanks

**Comment by KevinMusgrave** [comment]
Do you mean a jupyter notebook showing how to use VICRegLoss?

**Comment by stereomatchingkiss** [comment]
> Do you mean a jupyter notebook showing how to use VICRegLoss?

Yes, is it possible? Hard to know how to use it just look at the doc.

**Comment by KevinMusgrave** [comment]
It's possible, but might be lower priority. I'll leave this issue open anyway in case someone wants to make a pull request to add an example notebook.

I think you'll find the official paper implementation helpful: https://github.com/facebookresearch/vicreg

**Comment by TKassis** [comment]
Thank you @KevinMusgrave for such a masterpiece of a library. I have a question relating to the VICRegLoss which I think is what is confusing @stereomatchingkiss as well. I read the paper but I'm a bit confused about it's implementation in PML. Why does the loss not follow the typical (embeddings, labels) arguments like most of the other losses. What is the reference embedding? Why can't that just be a label referring to another embedding?

**Comment by KevinMusgrave** [comment]
Thanks @TKassis 😄 

When the implementation was discussed (#372) it seemed like this loss would only ever be used as shown in Figure 1 of [the paper](https://arxiv.org/pdf/2105.04906.pdf):

![image](https://user-images.githubusercontent.com/2314891/209448969-6e69352a-3c98-4f2e-ae7e-3fdeb87424b8.png)

So `embeddings` and `ref_emb` are always for the same set of data, but with different data augmentations.

That said, I would prefer if VICRegLoss followed the same format as the other loss functions. I've created a separate issue for this task now (#560).
